versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/files/FileExporter.java

```diff
@@ -39,23 +47,37 @@ public interface Builder {
 
   abstract Path targetDirectory();
 
+  @Override
+  @Nonnull
+  public Path getTargetPath() {
+    return targetDirectory();
+  }
+
   @Override
   public void preValidate() throws IOException {
-    if (Files.isDirectory(targetDirectory())) {
+    if (isDirectory(targetDirectory())) {
       try (Stream<Path> listing = Files.list(targetDirectory())) {
-        Preconditions.checkState(
+        checkState(
             !listing.findAny().isPresent(),
             "Target directory %s must be empty, but is not",
             targetDirectory());
       }
     } else {
-      Files.createDirectories(targetDirectory());
+      createDirectories(targetDirectory());
     }
   }
 
   @Override
-  public OutputStream newFileOutput(String fileName) throws IOException {
-    return Files.newOutputStream(targetDirectory().resolve(fileName));
+  @Nonnull
+  public OutputStream newFileOutput(@Nonnull String fileName) throws IOException {
+    checkArgument(
+        fileName.indexOf('/') == -1 && fileName.indexOf('\\') == -1, "Directories not supported");
```

Comment:
nit: would it make sense to do this check after a `Path` is constructed from the name so that we do not have to worry about OS-specific separators?

Reply:
\\ isn't recognized by Path on Unix (and vice versa)